### PR TITLE
LB-ALT-create-newList

### LINKED ISSUE #3: 3. As a user, I want to set up a new shopping list so I can track purchased items
A shopping list consists of a set of items associated with a user’s token. Creating a new list consists of the following:

* Generate a new, unique token
* Save the token to `localStorage`
* Show the user the list view

The following script can be used to generate a suitable token: https://gist.github.com/segdeha/21a42618ce5a54916c5b58d36ec2992e

AC:

* For users who do not already have a token/list, a button or link exists at the `/` route (i.e. home screen) that allows them to create a new list
* Clicking the button/link generates a new token and saves it to `localStorage`
* Once the token has been created and saved, the user is redirected to the "list" view

### PR DESCRIPTION
**Pull request by Lukabaramishvili**
## Description

For this issue, we created a new `Welcome` component that renders a button to generate new token and save it to the localStorage. The component checks if the user visits the app for the first time by checking the token's existence. If it is the first time, the user is presented with the button to create a new list. Otherwise redirected to the existing list. 

We also created a `token` file that generates tokens for users. 
Tokens are generated using the following approach found [here](https://gist.github.com/segdeha/21a42618ce5a54916c5b58d36ec2992e).

## Related Issue

Closes #3 

## Acceptance Criteria

- [x] For users who do not already have a token/list, a button or link exists at the / route (i.e. home screen) that allows them to create a new list
- [x] Clicking the button/link generates a new token and saves it to localStorage
- [x] Once the token has been created and saved, the user is redirected to the "list" view

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### After

Screen Shot 2020-06-10 at 5 54 57 PM

## Testing Steps / QA Criteria

- To test a new functionality, visit the landing page (`/` route) and press the `New List` button to create a new token. 
- Next, open your developer tools, navigate to Applications - > Local Storage, and see your new token.

### CODE REVIEW

**Comment by Lukabaramishvili** [comment]
Thanks for pointing that out. I believe the reason why the error pops up is that the initial value of `enteredValue` state is blank. You can verify it in the AddItem component (line: 6). Therefore, during the first rendering, this.state.enteredValue is undefined, which makes the input field uncontrolled. 
The solution here would be to initialize the `enteredValue` state with an empty string(''). However, since the `AddItem` component was repurposed by you and Lily, the error should be gone. In fact,`enteredValue` state is now initialized with the `emptyShoppingItem` object, which solves the issue.

**Comment by adriennefriend** [comment]
Happy Friday @lilyhoratio and @abidishajia ! Gentle encouragement to give this PR a re-review today, if possible, following @Lukabaramishvili 's comment! Of course, please feel encouraged to continue the conversation if needed! We're all in this together, building this together, and we want to make sure we have a clear, shared sense of the work.

File: src/components/Welcome.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  const push = history.push;
+
+  const handleClick = () => {
+    const token = GetToken();
+    saveToken(token);
+    push('/list');
+  };
+
+  const checkForToken = () => {
+    return localStorage.getItem('userToken') ? true : false;
```
[the remark this answers is not in the document]
Comment (Lukabaramishvili) [reply src/components/Welcome.js:17]: 💡Thanks for sharing this! I never used it before. Will try to incorporate in the future 🙂

**Comment by Lukabaramishvili** [comment]
@lilyhoratio Yes, I think we should definitely protect the route if the token does not exist. We can discuss as I group how we want to structure the code. Currently, Navbar is rendered in `App.js`. One option is to create a separate navbar component that will be rendered only in case the token exists.

**Pull request merged**